contrib/nitiwiki/src/wiki_base.nit

```diff
@@ -209,6 +209,26 @@ class Nitiwiki
 		return tpl
 	end
 
+	# Does a sideblock named `name` exists for this wiki?
+	fun has_sideblock(name: String): Bool do
+		name = "{name}.{config.md_ext}"
+		return expand_path(config.root_dir, config.sidebar_dir, name).file_exists
+	end
+
+	# Load a markdown block with `name` from `WikiConfig::sidebar_dir`.
+	private fun load_sideblock(name: String): nullable String do
+		if not has_sideblock(name) then
+			message("Error: can't load sideblock `{name}`", 0)
+			return null
+		end
+		name = "{name}.{config.md_ext}"
+		var path = expand_path(config.root_dir, config.sidebar_dir, name)
+		var file = new FileReader.open(path)
+		var res = file.read_all
+		file.close
```

Comment:
there is a oneliner `path.to_path.read_all` that does the opening/closing if I remember correclty